code/modules/mechs/equipment/combat_projectile.dm

```diff
@@ -0,0 +1,167 @@
+		list(mode_name="3-round bursts", burst=3,    fire_delay=null, one_hand_penalty=0,  burst_accuracy=list(0,-1,-1),   dispersion=list(0.0, 0.6, 1.0), autofire_enabled=0),
+		list(mode_name="full auto",      burst=1,    fire_delay=null, burst_delay=1,       one_hand_penalty=0,             burst_accuracy = list(0,-1,-1), dispersion=list(0.0, 0.6, 1.0), autofire_enabled=1)
+	)
+
+/obj/item/mech_equipment/mounted_system/projectile/machine
+	name = "mounted machine gun"
+	icon_state = "mech_machine_gun"
+	holding_type = /obj/item/gun/projectile/automatic/machine/mech
+	restricted_hardpoints = list(HARDPOINT_LEFT_HAND, HARDPOINT_RIGHT_HAND)
+	restricted_software = list(MECH_SOFTWARE_WEAPONS)
+
+/obj/item/gun/projectile/automatic/machine/mech
+	magazine_type = /obj/item/ammo_magazine/mech/rifle/drum
+	allowed_magazines = /obj/item/ammo_magazine/mech/rifle/drum
+	one_hand_penalty = 0
+	has_safety = FALSE
+	manual_unload = FALSE
+
+// Magazines below this.
+
+/obj/item/ammo_magazine/mech/attack_self(mob/user)
+	to_chat(user, SPAN_WARNING("It's pretty hard to extract ammo from a magazine that fits on a mech. You'll have to do it one round at a time."))
+	return
+
+/obj/item/ammo_magazine/mech/smg_top
+	name = "large 7mm magazine"
+	desc = "A large magazine for a mech's gun. Looks way too big for a normal gun."
+	icon_state = "smg_top"
+	mag_type = MAGAZINE
+	ammo_type = /obj/item/ammo_casing/pistol/small
+	material = /decl/material/solid/metal/steel
+	caliber = CALIBER_PISTOL_SMALL
+	max_ammo = 90
+
+/obj/item/ammo_magazine/mech/rifle
+	name = "large assault rifle magazine"
+	icon_state = "assault_rifle"
+	mag_type = MAGAZINE
+	caliber = CALIBER_RIFLE
+	material = /decl/material/solid/metal/steel
+	ammo_type = /obj/item/ammo_casing/rifle
+	max_ammo = 100
+
+/obj/item/ammo_magazine/mech/rifle/drum
+	name = "large machine gun magazine"
+	icon_state = "drum"
+	mag_type = MAGAZINE
+	caliber = CALIBER_RIFLE
+	material = /decl/material/solid/metal/steel
+	ammo_type = /obj/item/ammo_casing/rifle
+	max_ammo = 300
+
+// Handling for auto-fire mechanic
+/mob/living/exosuit/can_autofire(obj/item/gun/autofiring, atom/autofiring_at)
+	if(autofiring.autofiring_by != src)
+		return FALSE
+	var/client/C
+	if(current_user)
+		C = current_user.client
+		if(current_user.incapacitated())
+			return FALSE
+	else
+		C = client
+		if(incapacitated())
+			return FALSE
+	if(!C || !(autofiring_at in view(C.view, src)))
+		return FALSE
+	if(!(get_dir(src, autofiring_at) & dir))
+		return FALSE
+	if(!(autofiring in selected_system)) // Make sure the gun is still selected.
+		return FALSE
+	return TRUE
+
+/obj/item/mech_equipment/mounted_system/projectile/MouseDownInteraction(atom/object, location, control, params, mob/user)
+	var/obj/item/gun/gun = holding
+	if(user != src)
+		if(istype(object) && (isturf(object) || isturf(object.loc)) && !user.incapacitated() && istype(gun))
```

Comment:
Why are you checking these identical clauses twice rather than just checking them before `user != src`?